test/memory/uninitialized_move.cpp

```diff
@@ -16,25 +16,29 @@
 #include <stl2/detail/algorithm/count_if.hpp>
 #include <stl2/detail/algorithm/equal.hpp>
 #include <stl2/detail/memory/destroy.hpp>
+#include <stl2/iterator.hpp>
 #include <stl2/view/repeat.hpp>
 #include <stl2/view/take_exactly.hpp>
+#include <algorithm>
 #include <cstdint>
 #include <memory>
 #include <vector>
 #include "../simple_test.hpp"
 #include "common.hpp"
 
 namespace ranges = __stl2;
+using ranges::ext::span;
 
 namespace {
-	template <typename T>
-	bool empty(const Array<T>& a) {
-		return ranges::all_of(a, &T::empty);
+	template <ranges::Range Rng>
```

Comment:
This is underconstrained - should be something like:
```c++
template <ranges::InputRange Rng>
requires requires {
  typename ranges::value_type_t<Rng>;
  &ranges::value_type_t<Rng>::empty;
  requires ranges::Invocable<
    decltype(&ranges::value_type_t<Rng>::empty),
    ranges::reference_t<ranges::iterator_t<const Rng>>>;
}
```

Reply:
What is the difference between `requires Invocable<...>;` and `Invocable<...>;` here?

Reply:
Ah, I understand this now. Nested requirements are effectively the `typename` of concepts.

Reply:
Inside a requires expression:
* `requires Concept</*...*/>;` is a nested requirement that requires `Concept</*...*/>` to be satisfied. 
* `Concept</*...*/>` is an expression requirement that requires `Concept</*...*/>` to be a valid expression.

It's extremely unfortunate that the syntaxes are so similar: forgetting a `requires` on a nested requirement results in a valid program with totally different meaning.